MetaU Capstone Project/API/SpotifyManager.m

```diff
@@ -247,6 +260,144 @@ - (void)searchTrack:(NSString *)query type:(NSString *)type result:(void (^)(NSD
         NSData *dataUTF8 = [strISOLatin dataUsingEncoding:NSUTF8StringEncoding];
         dataRecieved = [NSJSONSerialization JSONObjectWithData:dataUTF8 options:0 error:&error];
         
+        if (dataRecieved != nil) {
+            parsingFinished([dataRecieved copy]);
+        } else {
+            NSLog(@"Error: %@", error.localizedDescription);
+            parsingFinished([[NSDictionary alloc] init]);
+        }
+    }] resume];
+}
+
+- (void)addQueueToSpotify: (NSString *)trackURI {
+    [[self.appRemote playerAPI] enqueueTrackUri:trackURI callback:^(id  _Nullable result, NSError * _Nullable error) {
+            if (error != nil) {
+                NSLog(@"Error: %@", error.localizedDescription);
+            } else {
+                NSLog(@"%@ added to Spotify queue", trackURI);
+            }
+    }];
+}
+
+- (NSDictionary *)currentTrackInfo: (id<SPTAppRemoteTrack>)track {
+    NSDictionary *trackDetails = @{
+            @"name" : track.name,
+            @"URI" : track.URI,
+            @"contextURI" : track.album.URI,
+            @"artist" : track.artist.name,
+            @"album" : track.album.name,
+            @"images" : @[]
+    };
+
+    return (NSDictionary *)trackDetails;
+}
+
+- (NSDictionary *)getCurrentTrack {
+    return self.currentTrack;
+}
+
+- (id<SPTAppRemoteTrack>)getCurrentTrackAsContentItem {
+    return self.currentTrackContentItem;
+}
+
+- (void)recentlyPlayedTrack {
+    NSString *targetURL = @"https://api.spotify.com/v1/me/player/recently-played?limit=1";
+    
+    [self retriveDataFrom:targetURL result:^(NSDictionary * _Nonnull dataRecieved) {
+        NSString *track = [[dataRecieved valueForKey:@"items"] valueForKey:@"track"][0];
+        
+        NSString *trackName = [track valueForKey:@"name"];
+        NSString *trackURI = [track valueForKey:@"uri"];
+
+        NSMutableArray *trackArtists = [[track valueForKey:@"artists"] valueForKey:@"name"];
+
+        NSString *trackAlbum = [[track valueForKey:@"album"] valueForKey:@"name"];
+        NSString *contextURI =  [[track valueForKey:@"album"] valueForKey:@"uri"];
+        NSArray *trackImages = [[[track valueForKey:@"album"] valueForKey:@"images"] valueForKey:@"url"];
```

Comment:
Consider using constants for some of these keys.

For example, I see that you have `@"name"` defined a couple times.
Using a constant would avoid the possibility of something being misspelled.